crates/ruff_linter/src/rules/mccabe/rules/function_is_too_complex.rs

```diff
@@ -96,8 +97,16 @@ fn get_complexity_number(stmts: &[Stmt]) -> usize {
                 complexity += get_complexity_number(orelse);
             }
             Stmt::Match(ast::StmtMatch { cases, .. }) => {
-                for case in cases {
+                let last_index = cases.len() - 1;
+                for (i, case) in cases.iter().enumerate() {
                     complexity += 1;
+                    if let Pattern::MatchAs(match_as_pattern) = &case.pattern {
+                        if match_as_pattern.pattern.is_none() && i == last_index {
```

Comment:
What's the reason that we restrict the logic to only when the `_` pattern is last? Can't we always subtract the complexity in that case?

Reply:
Because an [irrefutable pattern](https://peps.python.org/pep-0634/#irrefutable-case-blocks) can only occur in the last case block otherwise it's a syntax error (checked by the compiler). We don't raise this currently, but we should start doing so. I'm going to make an umbrella issue with such syntax errors raised by the compiler.

Reply:
That makes sense. But that also implies that it would be safe to ignore the check here and simply assume it is the last?